src/eos-updater-apply.c

```diff
@@ -103,14 +111,17 @@ apply (GTask *task,
     goto error;
 
   bootversion = ostree_sysroot_get_bootversion (sysroot);
-  merge_deployment = ostree_sysroot_get_merge_deployment (sysroot, NULL);
+  booted_deployment = eos_updater_get_booted_deployment_from_loaded_sysroot (sysroot,
+                                                                             &error);
+  if (booted_deployment == NULL)
+    goto error;
```

Comment:
for reference, otaylor wrote on this exact subject: https://blog.fishsoup.net/2015/11/05/attributecleanup-mixed-declarations-and-code-and-goto/